internal/services/springcloud/spring_cloud_dynatrace_application_performance_monitoring_resource_test.go

```diff
@@ -0,0 +1,175 @@
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_token", "tenant_token", "tenant"),
+	})
+}
+
+func TestAccSpringCloudDynatraceApplicationPerformanceMonitoringPerformanceMonitoring_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_dynatrace_application_performance_monitoring", "test")
+	r := SpringCloudDynatraceApplicationPerformanceMonitoringResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccSpringCloudDynatraceApplicationPerformanceMonitoringPerformanceMonitoring_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_dynatrace_application_performance_monitoring", "test")
+	r := SpringCloudDynatraceApplicationPerformanceMonitoringResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_token", "tenant_token", "tenant"),
+	})
+}
+
+func TestAccSpringCloudDynatraceApplicationPerformanceMonitoringPerformanceMonitoring_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_dynatrace_application_performance_monitoring", "test")
+	r := SpringCloudDynatraceApplicationPerformanceMonitoringResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_token", "tenant_token", "tenant"),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_token", "tenant_token", "tenant"),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_token", "tenant_token", "tenant"),
+	})
+}
+
+func (r SpringCloudDynatraceApplicationPerformanceMonitoringResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := appplatform.ParseApmID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.AppPlatform.AppPlatformClient.ApmsGet(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r SpringCloudDynatraceApplicationPerformanceMonitoringResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
```

Comment:
Shall we have leave a comment to explain why this is needed?

Reply:
Removed.